java/core/src/main/java/com/lancedb/lance/ReadOptions.java

```diff
@@ -71,8 +71,8 @@ public static class Builder {
 
     private Optional<Integer> version = Optional.empty();
     private Optional<Integer> blockSize = Optional.empty();
-    private int indexCacheSize = 256;
-    private int metadataCacheSizeBytes = 1024 * 1024 * 1024; // Default to 1 GiB like Rust
+    private long indexCacheSizeBytes = 6 * 1024 * 1024 * 1024; // Default to 6 GiB like Rust
```

Comment:
haven't gotten to the rust, but should we base the default sizing of a % of system RAM? That would allow us to ensure we don't OOM by default, right?

Reply:
Yeah I think that's a reasonable goal for the future.

Reply:
I'm just trying not to take on too much in one PR. I've already had this PR open for > 1 month. This PR allows the callers to finally be able to set as a % of system RAM if they want, something we couldn't do before. I think that's sufficient improvement. I'm happy to make a follow up PR that sets the defaults to be a % of RAM too.